SETUP.md

```diff
@@ -0,0 +1,78 @@
+# Contributing
+
+Thank you for considering contributing to the R Markdown C
+
+These instructions describe the steps and system requirements needed to build this book. 
+
+When contribution to the book, it is a good practice to build it locally to ensure that any modifications that you have made worked as expected. However, please do not include the built HTML or PDF in your pull request. It is easier to merge in plain text R, Markdown, and R Markdown files and to only reknit the book on the master branch. 
+
+## System Requirements
+
+### R Packages
+
+The following R packages are required to build to book. These are the same as those which are loaded in the `index.Rmd` file, plus `bookdown` and `tinytex`.
+
+```
+reqd_pkgs <-
+c('remotes', 'webshot', 'DiagrammeR', 'stargazer', 'gt', 'flextable', 'formatR',
+  'equatiomatic', 'gifski', 'magick', 'pdftools', 'tikzDevice', 'xaringan',
+  'kableExtra', 'blogdown', 'jsonlite', 'nomnoml', 'bookdown', 'tinytex')
+```
+
+Additionally, the following packages are referenced in the book. If you do not have them installed, the book will still build, but there references will be incomplete and you will receive numerous messages in the console of the form "pandoc-citeproc: reference R-{package name} not found". 
+
+```
+refd_pkgs <- 
+  c('animation', 'blastula', 'blogdown', 'bookdown','broom', 'Cairo', 'cairoDevice', 'condformat',
+    'dagitty', 'diagram', 'DiagrammeR', 'distill', 'downloadthis', 'DT', 'ezknitr', 'flair',
+    'flexdashboard', 'flextable', 'formatR', 'formattable', 'gganimate', 'ggdag', 'ggplot2', 
+    'gifski', 'googledrive', 'govdown', 'gt', 'gtsummary', 'here', 'huxtable', 'kableExtra', 
+    'knitcitations', 'knitr', 'learnr', 'magick', 'nomnoml', 'officedown', 'officer', 'pagedown',
+    'pander', 'pixiedust', 'pkgdown', 'printr', 'r2d3', 'reactable', 'reticulate', 'revealjs', 'rgl',
+    'rhandsontable', 'rmarkdown', 'rmdformats', 'roxygen2', 'rsconnect', 'rticles', 'sass', 'spelling', 
+    'stargazer', 'styler', 'svglite', 'tables', 'tangram', 'tikzDevice', 'tinytex', 'tufte', 'usethis',
+    'webshot', 'workflowr', 'xaringan', 'xfun', 'xtable', 'yaml', 'ztable')
+```
```

Comment:
Thanks! That's where I got the list. It will knit without them but just give warnings in the console. I tried to make it clear that these weren't necessary, but I'm happy to remove altogether if you think it is more confusing / burdensome.

Reply:
Yeah I think you're right. We should keep it as you explained. But I think we need to go further - because that means also for building our book we need these dependencies. So I guess we need to add an installation code for those. Otherwise we will have missing references.

Reply:
Overall, I really like your points about just leaving everything to install automatically for the benefit of the automation. I'll make these references like the `index.Rmd` and install automatically if missing if that is alright

Reply:
But you pointed a good point. We can't build the final book without those, so if we want to build the book automatically we need them. Right? 
So we need a mechanism to know that they are required for building the book in GHA. 
One of the aim is to have the book built automatically for next versions.

Reply:
Yeah I think that will help for our next versions. I hope we can build AND publish to bookdown.org automatically. I will do it for other books too.